src/main/java/com/conveyal/analysis/controllers/RegionalAnalysisController.java

```diff
@@ -406,17 +409,26 @@ private UrlWithHumanName getRegionalResults (Request req, Response res) throws I
         // For newer analyses that have multiple cutoffs, percentiles, or destination pointsets, these initial values
         // are coming from deprecated fields, are not meaningful and will be overwritten below from query parameters.
         int percentile = analysis.travelTimePercentile;
-        int cutoffMinutes = analysis.cutoffMinutes;
+        int threshold = analysis.cutoffMinutes;
         String destinationPointSetId = analysis.grid;
 
-        // Handle newer regional analyses with multiple cutoffs in an array.
         // If a query parameter is supplied, range check it, otherwise use the middle value in the list.
-        // The cutoff variable holds the actual cutoff in minutes, not the position in the array of cutoffs.
-        if (analysis.cutoffsMinutes != null) {
+        if (analysis.request.includeTemporalDensity) {
+            int nThresholds = analysis.request.dualAccessThresholds.length;
+            int[] thresholds = analysis.request.dualAccessThresholds;
+            checkState(nThresholds > 0, "Regional analysis has no thresholds.");
+            threshold = getIntQueryParameter(req, "threshold", thresholds[nThresholds / 2]);
+            checkArgument(new TIntArrayList(thresholds).contains(threshold),
+                    "Dual access thresholds for this regional analysis must be taken from this list: (%s)",
+                    Ints.join(", ", thresholds)
+            );
+        } else if (analysis.cutoffsMinutes != null) {
+            // Handle newer regional analyses with multiple cutoffs in an array.
+            // The cutoff variable holds the actual cutoff in minutes, not the position in the array of cutoffs.
             int nCutoffs = analysis.cutoffsMinutes.length;
             checkState(nCutoffs > 0, "Regional analysis has no cutoffs.");
-            cutoffMinutes = getIntQueryParameter(req, "cutoff", analysis.cutoffsMinutes[nCutoffs / 2]);
-            checkArgument(new TIntArrayList(analysis.cutoffsMinutes).contains(cutoffMinutes),
+            threshold = getIntQueryParameter(req, "cutoff", analysis.cutoffsMinutes[nCutoffs / 2]);
```

Comment:
I haven't thought this all the way through -- should all query parameters be `threshold` (regardless of whether the request is for accessibility or dual access results)?

```suggestion
            threshold = getIntQueryParameter(req, "threshold", analysis.cutoffsMinutes[nCutoffs / 2]);
```

Reply:
Yes. Updated in https://github.com/conveyal/r5/pull/955/commits/cb4334505b0f2237ae16555d9f574c545cb3af2c.